discord/interactions/disconnectCommand.ts

```diff
@@ -18,22 +18,26 @@ export const handleDisconnectConfirmCommand = async (
   const userId = interaction.member?.user?.id;
   const guildId = interaction.guildId;
   if (!userId || !guildId) return;
-  const alreadyDiscordMember = await DiscordMemberRepository.findOne({
+  const alreadyDiscordMember = await DiscordMemberRepository.find({
     where: {
       discordMemberId: userId,
-      discordServerId: guildId,
+      DiscordServerId: guildId,
     },
-    relations: ["discordServer"],
+    relations: ["DiscordServerConfig"],
   });
   if (!alreadyDiscordMember) return;
 
   await DiscordMemberRepository.softRemove(alreadyDiscordMember);
-  removeRole(
-    restClient,
-    guildId,
-    userId,
-    alreadyDiscordMember.discordServer.discordRoleId
-  );
+
+  for (let discordMember of alreadyDiscordMember) {
+    removeRole(
+      restClient,
+      guildId,
+      userId,
+      discordMember.DiscordServerConfig.discordRoleId
+    );
+  }
```

Comment:
same here we would probably have only one discordMember, but loop on all the discordMember.discordServer.configurations and remove the role for each configuration